dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/tracker/export/relationship/RelationshipsExportControllerTest.java

```diff
@@ -569,62 +567,79 @@ void getRelationshipsByTrackedEntityWithEnrollments() {
     assertHasOnlyMembers(enrollments.get(0), "trackedEntity", "enrollment");
   }
 
-  //  @Test
-  //  void getRelationshipsByTrackedEntityAndEnrollmentWithAttributesIsEmpty() {
-  //    // Tracked entity attribute values are owned by the tracked entity and only mapped onto the
-  //    // enrollment on export. Program tracked entity attributes are only returned by the
-  // underlying
-  //    // TE service if a program is
-  //    // provided which is not possible on the relationship endpoint.
-  //    TrackedEntity to = trackedEntity(orgUnit);
-  //    to.setTrackedEntityAttributeValues(
-  //        Set.of(attributeValue(tea, to, "12"), attributeValue(tea2, to, "24")));
-  //    program.setProgramAttributes(List.of(createProgramTrackedEntityAttribute(program, tea2)));
-  //    Enrollment from = enrollment(to);
-  //    Relationship relationship = relationship(from, to);
-  //
-  //    RelationshipType type = relationship.getRelationshipType();
-  //
-  //    RelationshipConstraint fromConstraint = new RelationshipConstraint();
-  //
-  //    TrackerDataView trackerDataView = new TrackerDataView();
-  //    trackerDataView.setAttributes(new LinkedHashSet<>(Set.of(tea2.getUid())));
-  //
-  //    fromConstraint.setTrackerDataView(trackerDataView);
-  //
-  //    RelationshipConstraint toConstraint = new RelationshipConstraint();
-  //
-  //    TrackerDataView dataView = new TrackerDataView();
-  //    dataView.setAttributes(new LinkedHashSet<>(Set.of(tea.getUid(), tea2.getUid())));
-  //
-  //    toConstraint.setTrackerDataView(dataView);
-  //
-  //    type.setFromConstraint(fromConstraint);
-  //    type.setToConstraint(toConstraint);
-  //    switchContextToUser(user);
-  //
-  //    JsonList<JsonRelationship> relationships =
-  //        GET(
-  //
-  // "/tracker/relationships?trackedEntity={trackedEntity}&fields=from[enrollment[attributes[attribute,value]]],to[trackedEntity[attributes[attribute,value]]]",
-  //                to.getUid())
-  //            .content(HttpStatus.OK)
-  //            .getList("relationships", JsonRelationship.class);
-  //
-  //    JsonList<JsonAttribute> enrollmentAttr =
-  //        relationships.get(0).getFrom().getEnrollment().getAttributes();
-  //    assertIsEmpty(
-  //        enrollmentAttr.toList(JsonAttribute::getAttribute),
-  //        "program attributes should not be returned as no program can be provided");
-  //    JsonList<JsonAttribute> teAttributes =
-  //        relationships.get(0).getTo().getTrackedEntity().getAttributes();
-  //    assertContainsAll(List.of(tea.getUid()), teAttributes, JsonAttribute::getAttribute);
-  //    assertContainsAll(List.of("12"), teAttributes, JsonAttribute::getValue);
-  //  }
-  //
-
-  @Test
-  void getRelationshipsByTrackedEntityWithProgramOwners() {
+  @Test
+  void shouldGetRelationshipsByTrackedEntityWithAttributes() {
```

Comment:
is there a test using the same request without tracker data view to show what influence it has?

Reply:
I added the tests without tracker data view.